panic: runtime error: index out of range

The query doesn't make sense but it shouldn't cause a panic.

```
from(bucket: "telegraf/autogen")
  |> range(start: dashboardTime)
  |> filter(fn: (r) => r._measurement == "apache")
  |> covariance(columns: ["column_x", "column_y"], pearsonr: false, valueDst: "_value")
  |> set(key: "myKey", value: "myValue")
```

```
panic: runtime error: index out of range
goroutine 17895 [running]:
runtime/debug.Stack(0x1c78b60, 0x1c682c0, 0x2903cc0)
	/usr/local/Cellar/go/1.11.1/libexec/src/runtime/debug/stack.go:24 +0xa7
github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute.(*poolDispatcher).Start.func1.1(0xc0000a1440)
	/Users/rsavage/go/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute/dispatcher.go:73 +0xbc
panic(0x1c682c0, 0x2903cc0)
	/usr/local/Cellar/go/1.11.1/libexec/src/runtime/panic.go:513 +0x1b9
github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/functions/transformations.(*CovarianceTransformation).Process(0xc000b44960, 0x215e08f846dbaef9, 0xdd3ebf055b48b1bc, 0x5ac8c58, 0xc0002365a0, 0x103ec01, 0xc0000a0b98)
	/Users/rsavage/go/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/functions/transformations/covariance.go:177 +0x5d8
github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute.processMessage(0x1f5e1a0, 0xc000b44960, 0x1f4cfa0, 0xc0002a9e80, 0xc0012ce768, 0x11eb53d, 0xc0008ca590)
	/Users/rsavage/go/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute/transport.go:199 +0x1fc
github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute.(*consecutiveTransport).processMessages(0xc0000c6300, 0xa)
	/Users/rsavage/go/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute/transport.go:156 +0xa1
github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute.(*consecutiveTransport).processMessages-fm(0xa)
	/Users/rsavage/go/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute/transport.go:137 +0x34
github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute.(*poolDispatcher).run(0xc0000a1440, 0x1f5a460, 0xc000759d10)
	/Users/rsavage/go/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute/dispatcher.go:126 +0x4b
github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute.(*poolDispatcher).Start.func1(0xc0000a1440, 0x1f5a460, 0xc000759d10)
	/Users/rsavage/go/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute/dispatcher.go:80 +0x95
created by github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute.(*poolDispatcher).Start
	/Users/rsavage/go/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/flux/execute/dispatcher.go:61 +0x7e
```

--- a/functions/transformations/covariance.go
+++ b/functions/transformations/covariance.go
@@ -171,7 +171,13 @@ func (t *CovarianceTransformation) Process(id execute.DatasetID, tbl flux.Table)
 		return err
 	}
 	xIdx := execute.ColIdx(t.spec.Columns[0], cols)
+	if xIdx < 0 {
+		return fmt.Errorf("specified column does not exist in table: %v", t.spec.Columns[0])
+	}
 	yIdx := execute.ColIdx(t.spec.Columns[1], cols)
+	if yIdx < 0 {
+		return fmt.Errorf("specified column does not exist in table: %v", t.spec.Columns[1])
+	}
 
 	if cols[xIdx].Type != cols[yIdx].Type {
 		return errors.New("cannot compute the covariance between different types")

--- a/functions/transformations/testdata/covariance_missing_column_1.flux
+++ b/functions/transformations/testdata/covariance_missing_column_1.flux
@@ -1,0 +1,6 @@
+from(bucket: "test")
+	|> range(start: 2018-05-22T19:53:26Z)
+  |> covariance(columns: ["x", "y"])
+	|> yield(name: "0")
+
+

--- a/functions/transformations/testdata/covariance_missing_column_1.in.csv
+++ b/functions/transformations/testdata/covariance_missing_column_1.in.csv
@@ -1,0 +1,8 @@
+#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,double,string
+#group,false,false,false,false,false,false,true
+#default,_result,,,,,,
+,result,table,_start,_stop,_time,x,_measurement
+,,0,2018-05-22T19:53:24.421470485Z,2018-05-22T19:54:24.421470485Z,2018-05-22T19:53:26Z,0,cpu
+,,0,2018-05-22T19:53:24.421470485Z,2018-05-22T19:54:24.421470485Z,2018-05-22T19:53:36Z,0,cpu
+,,0,2018-05-22T19:53:24.421470485Z,2018-05-22T19:54:24.421470485Z,2018-05-22T19:53:46Z,2,cpu
+,,0,2018-05-22T19:53:24.421470485Z,2018-05-22T19:54:24.421470485Z,2018-05-22T19:53:56Z,7,cpu

--- a/functions/transformations/testdata/covariance_missing_column_1.out.csv
+++ b/functions/transformations/testdata/covariance_missing_column_1.out.csv
@@ -1,0 +1,5 @@
+#datatype,string,string
+#group,true,true
+#default,,
+,error,reference
+,specified column does not exist in table: y,

--- a/functions/transformations/testdata/covariance_missing_column_2.flux
+++ b/functions/transformations/testdata/covariance_missing_column_2.flux
@@ -1,0 +1,6 @@
+from(bucket: "test")
+	|> range(start: 2018-05-22T19:53:26Z)
+  |> covariance(columns: ["x", "y"])
+	|> yield(name: "0")
+
+

--- a/functions/transformations/testdata/covariance_missing_column_2.in.csv
+++ b/functions/transformations/testdata/covariance_missing_column_2.in.csv
@@ -1,0 +1,8 @@
+#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,double,string
+#group,false,false,false,false,false,false,true
+#default,_result,,,,,,
+,result,table,_start,_stop,_time,y,_measurement
+,,0,2018-05-22T19:53:24.421470485Z,2018-05-22T19:54:24.421470485Z,2018-05-22T19:53:26Z,0,cpu
+,,0,2018-05-22T19:53:24.421470485Z,2018-05-22T19:54:24.421470485Z,2018-05-22T19:53:36Z,0,cpu
+,,0,2018-05-22T19:53:24.421470485Z,2018-05-22T19:54:24.421470485Z,2018-05-22T19:53:46Z,2,cpu
+,,0,2018-05-22T19:53:24.421470485Z,2018-05-22T19:54:24.421470485Z,2018-05-22T19:53:56Z,7,cpu

--- a/functions/transformations/testdata/covariance_missing_column_2.out.csv
+++ b/functions/transformations/testdata/covariance_missing_column_2.out.csv
@@ -1,0 +1,5 @@
+#datatype,string,string
+#group,true,true
+#default,,
+,error,reference
+,specified column does not exist in table: x,

